File: src/main/java/org/calvin/__Book/pie/4-linked-lists/ListElement.java
public class ListElement<T> {
    private ListElement<T> next;
    private T              data;

    public ListElement(T value) {
        data = value;
    }

    public ListElement<T> next() {
        return next;
    }

    public T value() {
        return data;
    }

    public void setNext(ListElement<T> elem) {
        next = elem;
    }

    public void setValue(T value) {
        data = value;
    }
}

File: src/main/java/org/calvin/__Book/pie/4-linked-lists/ListElement.cpp
# include <iostream>

template <class T>
class ListElement {
    public:
        ListElement(T value): next(NULL), data(value) {
        }
        ~ListElement() {
        }

        ListElement* getNext() const {
            return next;
        }
        const T& value() {
            return data; 
        }
        void setNext(ListElement* elem) {
            next = elem; 
        }
        void setValue(const T &value) {
            data = value; 
        }
    private:
        ListElement *next;
        T           data;
};
